src/components/TagDropdown.tsx

```diff
@@ -26,14 +26,13 @@ export default function TagDropdown(props: TagDropdownProps) {
     const [searchInput, setSearchInput] = useState("");
     const [filteredTags, setFilteredTags] = useState(props.tags);
 
-    const handleSearch = (event) => {
+    const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const input_text: string = event.target.value;
         event.preventDefault();
-        setSearchInput(event.target.value);
-        // console.log(searchInput);
-        if (searchInput.length > 0) {
-            setFilteredTags(props.tags.filter((tag) => { return tag.match(searchInput) }));
+        setSearchInput(input_text);
+        if (input_text.length > 0) {
+            setFilteredTags(props.tags.filter((tag) => { return tag.match(input_text) }));
         } else {
-            // console.log("logging at " + searchInput);
             setFilteredTags(props.tags);
         }
     };
```

Comment:
The important change here is that we want to filter our list based on the input that we just got from the event, not from the stateful variable. Calling `setSearchInput` doesn't immidiately update `searchInput`, so we were filtering on the **previous** search text.

If we use a raw typescript variable to contain the search text, it'll work the way we want!